pyiron_atomistics/interactive/scipy_minimizer.py

```diff
@@ -291,24 +292,84 @@ def calc_minimize(
-            reliably)
-        ionic_steps (int): max number of steps
-        ionic_force_tolerance (float): maximum force tolerance
-    """
-
-    def __init__(self, input_file_name=None, table_name="input"):
-        self.minimizer = "CG"
-        self.ionic_steps = 100
-        self.ionic_force_tolerance = 1.0e-2
-        self.pressure = None
-        self.volume_only = False
-        self.ionic_energy_tolerance = 0
-        self.pressure_tolerance = 1.0e-3
-
+class Input(HasStorage):
+
+    def __init__(self):
+        super().__init__()
+        self.storage.table_name = "parameters"
+        self.storage.minimizer = "CG"
+        self.storage.ionic_steps = 100
+        self.storage.ionic_energy_tolerance = 0
+        self.storage.ionic_force_tolerance = 1.0e-2
+        self.storage.pressure_tolerance = 1.0e-3
+        self.storage.pressure = None
+        self.storage.volume_only = False
+
+    @property
+    def minimizer(self) -> str:
+        """str: name of minimizer to use"""
+        return self.storage.minimizer
+
+    @minimizer.setter
+    def minimizer(self, value: str):
+        self.storage.minimizer = value
+
+    @property
+    def ionic_steps(self) -> int:
+        """int: maximum number of minimization steps"""
+        return self.storage.ionic_steps
+
+    @ionic_steps.setter
+    def ionic_steps(self, value: int):
+        self.storage.ionic_steps = value
+
+    @property
+    def ionic_force_tolerance(self) -> float:
+        """float: convergence goal in terms of forces"""
+        return self.storage.ionic_force_tolerance
+
+    @ionic_force_tolerance.setter
+    def ionic_force_tolerance(self, value: float):
+        return self.storage.ionic_force_tolerance
+
+    @property
+    def ionic_energy_tolerance(self) -> float:
+        """float: convergence goal in terms of energye"""
+        return self.storage.ionic_energy_tolerance
+
+    @ionic_energy_tolerance.setter
+    def ionic_energy_tolerance(self, value: float):
+        self.storage.ionic_energy_tolerance = value
+
+    @property
+    def pressure_tolerance(self) -> float:
+        """float: convergence goal in terms of energye"""
+        return self.storage.pressure_tolerance
+
+    @pressure_tolerance.setter
+    def pressure_tolerance(self, value: float):
+        self.storage.pressure_tolerance = value
+
+    @property
+    def pressure(self) -> List[float]:
+        """float: target pressure"""
+        return self.storage.pressure
+
+    @pressure.setter
+    def pressure(self, value: Iterable[float]):
+        value = list(value)
+        if len(value) != 6:
+            raise ValueError(f"value must be length 3 iterable not: {value}!")
+        self.storage.pressure = value
+
+    @property
+    def volume_only(self) -> bool:
+        """bool: only pressure minimization"""
+        return self.storage.volume_only
+
+    @volume_only.setter
+    def volume_only(self, value: bool):
+        self.storage.volume_only = value
```

Comment:
I thought when using the data container, we do not have to define the individual properties, so I am a bit surprised you still do it here manually. Maybe we can use the set attribute functionality to reduce this redundancy.

Reply:
Using `DataContainer` subclasses directly causes a lot of trouble if one does not pay attention to the recursive data structure (For that reason this Input subclass was not working as intended). `HasStorage`, however, has everything inside the `storage` attribute in the end and we need to pull this up one level... We had quite some discussion in https://github.com/pyiron/pyiron/issues/1412 how to handle these things nicely. However, so far no one has implemented the corresponding class.

Reply:
Another reason I did it manually is to give well defined location to add input validation.  (I had tried this with the pressure attribute, but got it wrong and I don't have time to do it properly.)